bpf/bpf_lxc.c

```diff
@@ -337,6 +337,12 @@ static __always_inline int ipv6_l3_from_lxc(struct __ctx_buff *ctx,
 		send_trace_notify(ctx, TRACE_TO_HOST, SECLABEL, HOST_ID, 0,
 				  HOST_IFINDEX, reason, monitor);
 
+		cilium_dbg_capture(ctx, DBG_CAPTURE_DELIVERY, HOST_IFINDEX);
+		return redirect(HOST_IFINDEX, BPF_F_INGRESS);
+	} else if (is_defined(ENABLE_HOST_FIREWALL) && *dstID == HOST_ID) {
+		send_trace_notify(ctx, TRACE_TO_HOST, SECLABEL, HOST_ID, 0,
+				  HOST_IFINDEX, reason, monitor);
```

Comment:
nit for readability: We could probably avoid duplicating the trace + redirect by splitting the logic:

```
if (is_defined(HOST_REDIRECT_TO_INGRESS)) {
   ... perform IPv6 l3
   redirect = true
} else if (is_defined(ENABLE_HOST_FIREWALL) && *dstID == HOST_ID) {
   redirect = true
}
if (redirect) {
   ....send_trace_notify() / return redirect()
}
```

(Not a big deal but I had to stare at this for a while to convince myself that the various parameters to the trace/redirect are identical)